File: src/components/augment_info/elemental.ts
import { Augment, EFFECTS as eff } from "./_base";

const GROUP = "ELEMENTAL";

export let elemental: Array<Array<Augment>> = [];

let exploit: Array<Augment> = [];

const ELE_AUG = ["fire", "ice", "lightning", "wind"];
ELE_AUG.forEach((ele) => {
    exploit.push(
        new Augment(
            `${ele} exploit`,
            1,
            {
                [eff.MEL_POTENCY]: 1.025,
                [eff.RNG_POTENCY]: 1.025,
                [eff.TEC_POTENCY]: 1.025,
            },
            6,
            GROUP,
        ),
    );
});
elemental.push(exploit);

File: _data/augments/elemental.py
from typing import List

from ._augment import Augment
from ._augment_groups import AugmentGroups
from effect import *
from util import many_effs_with_same_amount

GROUP = AugmentGroups.ELE
CONFLICT = (GROUP,)

augments: List[Augment] = []

# -------------------------------------------
_eles = ("fire", "ice", "lightning", "wind", "light")
for ele in _eles:
    augments.append(
        Augment(
            f"{ele} exploit",
            1,
            6,
            many_effs_with_same_amount(
                OFFENSIVE_POT,
                1.025,
            ),
            GROUP,
            CONFLICT,
            f"against enemies weak to {ele}.",
        )
    )
# -------------------------------------------
